packages/core/src/Cardano/types/Transaction.ts

```diff
@@ -82,9 +81,29 @@ export interface Redeemer {
 
 export type Signatures = Map<Ed25519PublicKey, Ed25519Signature>;
 
-export type Witness = Omit<Partial<Ogmios.TxAlonzo['witness']>, 'redeemers' | 'signatures'> & {
+export declare type Signature = string;
+export declare type ChainCode = string;
+export declare type AddressAttributes = string;
+export declare type VerificationKey = string;
+export declare type Datum = string;
+
+export interface BootstrapWitness {
+  signature?: Signature;
+  chainCode?: ChainCode | undefined;
+  addressAttributes?: AddressAttributes | undefined;
+  key?: VerificationKey;
+}
+
+export type Witness = {
   redeemers?: Redeemer[];
   signatures: Signatures;
+  scripts?: {
+    [k: string]: Cardano.Script;
```

Comment:
Let's use type aliases for indexer types so that it's clear what it is

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/388/commits/f93bd1a2bac1896cc1289c422db283dd6dfd92dd

Reply:
Sorry this is not what I meant here. More like:

```
[k: WhateverThisIs]: Cardano.Script
```

Do you think having separate fields for script types is nicer? I guess it depends on how it would be used. Consider something like this:

```ts
type ScriptPlutusV2 = { __type: 'plutusv2', bytes: Cardano.HexBlob; }
type Script = ScriptNative | ScriptPlutusV2 | ...`.
```

Reply:
I am not sure why they are typified like that in Ogmios:

`[k: string]: Cardano.Script;`

Scripts are stored in the witness set in plain arrays (that's why I changed that type to just Script[]), each type of script; Native, PlutusV1 and PlutusV2 are stored in three different arrays inside the witness set:

https://github.com/input-output-hk/cardano-ledger/blob/b24fa846e960ff1e78d73b4a2d21e7c6a59a20dc/eras/babbage/test-suite/cddl-files/babbage.cddl#L275

I split them to keep our Witness type consistent with the type defined in the CDDL; however; I think merging them in a Script type also works, I will refactor this.

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/388/commits/e5cd2cc64cabb6553678c4661aab7639650ca9be